terraform/modules/antivirus-sns/subscriptions.tf

```diff
@@ -0,0 +1,6 @@
+resource "aws_sns_topic_subscription" "antivirus_to_s3_file_upload_notification_topic" {
+  topic_arn              = "${aws_sns_topic.s3_file_upload_notification.arn}"
+  protocol               = "https"
+  endpoint               = "https://${var.antivirus_api_basic_auth}@${var.antivirus_api_url}/callback"
```

Comment:
As I said, miiiight be nicer to have this as `antivirus_api_host` rather than `antivirus_api_url` as it's clearer and just accept we're going to handle all the path stuff here.